libfreerdp/core/settings.c

```diff
@@ -471,7 +472,17 @@ rdpSettings* freerdp_settings_new(DWORD flags)
 		settings->HomePath = GetKnownPath(KNOWN_PATH_HOME);
 		if (!settings->HomePath)
 				goto out_fail;
-		settings->ConfigPath = GetKnownSubPath(KNOWN_PATH_XDG_CONFIG_HOME, "freerdp");
+		base = GetKnownSubPath(KNOWN_PATH_XDG_CONFIG_HOME,
+				       FREERDP_VENDOR_STRING);
+		if (base)
+		{
+			settings->ConfigPath = GetCombinedPath(base, FREERDP_PRODUCT_STRING);
+			if (!PathFileExistsA(base))
+				if (!CreateDirectoryA(base, NULL))
+					settings->ConfigPath = NULL;
+		}
+		free (base);
```

Comment:
This would change the default configuration location (~/.config/freerdp) on linux to ~/.config/FreeRDP/FreeRDP as well ;(. Please make sure that the current path doesn't change for compatibility.